File: Assignment_41/Program5.c
/*
Write a recursive program which display below pattern
Input : 6
Output : a b c d e f
*/

#include <stdio.h>

void Display(int iNo)
{
    static char ch = 'a';

    if(iNo < 0)
    {
        iNo = -iNo;
    }
    
    if(iNo >= 1)          
    {
        printf(" %c\t",ch);
        ch++;
        Display(iNo-1);
    }
}

int main()
{
    int iValue = 0;

    printf("Enter number : ");
    scanf("%d",&iValue);

    Display(iValue);

    return 0;
}

File: Assignment_49/Program5.java
/* Write a java program which accepts string from user and display below pattern.
Input :  Hello
Output : H  #  #   #  #   
         H  e  #   #  #
         H  e  l   #  #
         H  e  l   l  #
         H  e  l   l  o
         H  e  l   l  #
         H  e  l   #  #
         H  e  #   #  #
         H  #  #   #  #
*/

import java.util.*;

class Pattern
{
    public void Pattern(String str)
    {
        int i = 0,j = 0;

        for(i = 0;i < str.length();i++)
        {
            for(j = 0;j <= str.length() - 1;j++)
            {
                if(i >= j)
                {
                    System.out.print(str.charAt(j)+"\t");
                }
                else
                {
                    System.out.print("#\t");
                }
            }
            System.out.println();
        }

        for(i = str.length() - 2;i >= 0 ;i--)
        {
            for(j = 0;j < str.length();j++)
            {
                if(i >= j)
                {
                    System.out.print(str.charAt(j)+"\t");
                }
                else
                {
                    System.out.print("#\t");
                }
            }
            System.out.println();
        }
    }
}

class Program5
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the String : ");
        String str = sobj.nextLine();

        Pattern pobj = new Pattern();
        pobj.Pattern(str);
    }
}